lib/burnside/bridge.cc

```diff
@@ -458,9 +462,39 @@ class FIRConverter {
     build().create<M::CallOp>(toLocation(), callee, operands);
     build().create<fir::UnreachableOp>(toLocation());
   }
-  void genFIR(const Pa::ReturnStmt &stmt) {
-    build().create<M::ReturnOp>(toLocation());  // FIXME: argument(s)?
+  void genReturnStmt(AnalysisData &, const Pa::FunctionSubprogram &func) {
+    auto &stmt{std::get<Pa::Statement<Pa::FunctionStmt>>(func.t)};
+    auto &name{std::get<Pa::Name>(stmt.statement.t)};
+    assert(name.symbol);
```

Comment:
does this consider the _result-name_ binding?

Reply:
Yes, name resolve is setting up the result symbol (`&details.result` below) accordingly, so it is transparent here.